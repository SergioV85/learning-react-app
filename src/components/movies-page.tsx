import * as React from 'react';

import { ErrorBoundary } from './error-boundary/error-boundary.component';
import { Footer } from './footer/footer.component';
import { Header } from './header/header.component';
import { IMovieObject } from './movie-card/movie-card.component';
import { MovieDetails } from './movie-details/movie-details.component';
import { MoviesList } from './movies-list/movies-list.component';
import { SearchPanel } from './search-panel/search-panel.component';
import { StatusBarPanel } from './status-bar/status-bar.component';

export interface IMoviesPageProps {
  movies: {
    list: IMovieObject[];
    selectedMovie: IMovieObject | null;
  };
  search: {
    searchBy: string;
    search: string;
    sortBy: string;
    sortOrder: string;
  };
  backToList(): void;
  changeSearchInput(): void;
  changeSearchType(): void;
  changeSortingOrder(): void;
  changeSortingType(): void;
  searchMovies(): void;
  selectMovie(id: number): void;
}

export class MoviesPage extends React.Component<IMoviesPageProps, {}> {
  constructor(props: IMoviesPageProps) {
    super(props);
  }
  public render() {
    return (
    <div className='c-page'>
      <div className='c-page__header'>
        <div className='c-page__header-shadow'>
          <div className='p-2 container'>
            <Header
                showBackButton={this.props.movies.selectedMovie !== null}
                onBackToList={this.props.backToList}
            />
            {this.panelOrDetails}
          </div>
        </div>
      </div>
      <div className='c-page__status-bar'>
        <StatusBarPanel
            foundMovies={this.props.movies.list.length}
            currentSortingOrder={this.props.search.sortOrder}
            currentSortingType={this.props.search.sortBy}
            onChangeSortingOrder={this.props.changeSortingOrder}
            onChangeSortingType={this.props.changeSortingType}
        />
      </div>
      <div className='c-page__movies'>
        <ErrorBoundary key='movies-list-wrapper' errorMessage='Unfortunately, we cant show movies list'>
          <MoviesList
              movies={this.props.movies.list}
              onMovieClick={this.props.selectMovie}
          />
        </ErrorBoundary>
      </div>
      <div className='c-page__footer'>
        <div className='p-2 container'>
          <Footer />
        </div>
      </div>
    </div>
    );
  }

  private get panelOrDetails() {
    return this.props.movies.selectedMovie !== null
      ? <ErrorBoundary
            key='movie-details-wrapper'
            errorMessage='Unfortunately, error happens during showing details of selected movie'
      >
          <MovieDetails selectedMovie={this.props.movies.selectedMovie} />
      </ErrorBoundary>
      : <ErrorBoundary
            key='search-panel-wrapper'
            errorMessage='Unfortunately, we can not show search panel'
      >
          <SearchPanel
              searchType={this.props.search.searchBy}
              search={this.props.search.search}
              onChangeType={this.props.changeSearchType}
              onInputChange={this.props.changeSearchInput}
              onSearchMovies={this.props.searchMovies}
          />
      </ErrorBoundary>;
  }
}
