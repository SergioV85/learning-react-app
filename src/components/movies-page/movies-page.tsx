import * as React from 'react';

import { ErrorBoundary } from './../error-boundary/error-boundary.component';

import { IMovieObject } from './../movie-details/movie-details.component';
import { MoviesEmptyList } from './../movies-empty-list/movies-empty-list.component';
import { MoviesList } from './../movies-list/movies-list.component';

export interface IMoviesPageProps {
  movies: IMovieObject[];
  selectMovie(id: number): void;
}

export class MoviesPage extends React.Component<IMoviesPageProps, {}> {
  constructor(props: IMoviesPageProps) {
    super(props);
  }
  public render() {
    return (
      <div className='c-page__movies'>
        {this.moviesListOrMessage}
      </div>
    );
  }

  private get moviesListOrMessage() {
    return this.props.movies.length !== 0
      ? <ErrorBoundary key='movies-list-wrapper' errorMessage='Unfortunately, we cant show movies list'>
          <MoviesList
              movies={this.props.movies}
              onMovieClick={this.props.selectMovie}
          />
        </ErrorBoundary>
      : <ErrorBoundary key='movies-empty-message-wrapper' errorMessage='Unfortunately, we cant show message'>
          <MoviesEmptyList />
        </ErrorBoundary>;
  }
}
