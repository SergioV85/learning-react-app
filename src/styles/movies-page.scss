@import './variables';

.c-page {
  display: flex;
  flex-direction: column;
  height: 100%;

  &__header {
    background-image: url('~assets/header_bg.jpg');
    background-position: 0 0;
    background-repeat: no-repeat;
    background-size: cover;

    &-shadow {
      background-color: rgba($color: #000000, $alpha: 0.75);
      height: 100%;
      width: 100%;
    }
  }
  &__status-bar {
    background-color: $primary-grey;
    min-height: 50px;
  }
  &__movies {
    flex: 1;
  }
  &__footer {
    background-color: $dark-grey;
  }
}