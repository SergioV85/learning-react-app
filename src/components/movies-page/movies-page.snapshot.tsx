import { shallow } from 'enzyme';
import * as React from 'react';
import { MoviesPage } from './movies-page';

const mockedStore = {
  movies: [],
};
const mockedProps = {
  selectMovie: jest.fn(),
};

describe('Movies Page Snapshot', () => {
  it('should render default page', () => {
    const initMoviesPage = shallow(<MoviesPage
      movies={mockedStore.movies}
      selectMovie={mockedProps.selectMovie}
    />);
    expect(initMoviesPage)
      .toMatchSnapshot();
  });
});
