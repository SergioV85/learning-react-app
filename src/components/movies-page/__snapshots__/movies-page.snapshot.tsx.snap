// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Page Snapshot should render default page 1`] = `
<div
  className="c-page__movies"
>
  <ErrorBoundary
    errorMessage="Unfortunately, we cant show message"
    key="movies-empty-message-wrapper"
  >
    <Component />
  </ErrorBoundary>
</div>
`;
